chromebook-setup.sh

```diff
@@ -655,6 +689,103 @@ cmd_eject_storage()
     echo "All done."
 }
 
+# -----------------------------------------------------------------------------
+# Experimental: Create Fedora images for Chromebooks
+cmd_setup_fedora_rootfs()
+{
+    local loopdev
+    local image
+    local btrfs
+
+    image=$IMAGE
+    loopdev="$(sudo losetup --show -fP $image)"
+    btrfs="${image/raw/btrfs}"
+    sudo dd if="${loopdev}p3" of="/var/tmp/$btrfs" conv=fsync status=progress
+    sudo losetup -D
+    mkdir ./tmpdir
+    sudo mount "/var/tmp/$btrfs" ./tmpdir
+    sleep 3
+    echo "Disable SELINUX"
+    sudo sed -i 's/SELINUX=enforcing/SELINUX=permissive/' ./tmpdir/root/etc/selinux/config
+
+    echo "remove root password"
+    sudo sed -i 's/root:!locked:/root:/' ./tmpdir/root/etc/shadow
+
+    echo "modifying fstab"
+    sudo sed -i '1,14s/^[^#]/# &/g' ./tmpdir/root/etc/fstab
+    sudo sed -i \
+    -e '/home/s/home//' \
+    -e '/home/s/btrfs/ext4/' \
+    -e 's/subvol=home,compress=zstd:1/defaults/' ./tmpdir/root/etc/fstab
+
+    # Copy the ROOTFS to media
+    echo "copying ROOTFS to partition"
+    sudo cp -ar "./tmpdir/root/"* "$ROOTFS_DIR"
+    sudo umount ./tmpdir
+    sudo rm -rf ./tmpdir
+
+    echo "Done."
+}
+
+cmd_setup_fedora_kernel()
+{
+    local kernel_version="5.16.2-0.test.fc35.aarch64"
+
+    # Download a known kernel that works for Chromebooks
+    [ -f kernel-core-$kernel_version.rpm ] || curl -OL https://download.copr.fedorainfracloud.org/results/eballetbo/fedora/fedora-35-aarch64/03507242-kernel/kernel-core-$kernel_version.rpm
+    [ -f kernel-modules-$kernel_version.rpm ] || curl -OL https://download.copr.fedorainfracloud.org/results/eballetbo/fedora/fedora-35-aarch64/03507242-kernel/kernel-modules-$kernel_version.rpm
+
+    # Extract and copy the kernel packages to the rootfs
+    mkdir ./tmpdir && cd ./tmpdir
+    rpm2cpio ../kernel-core-$kernel_version.rpm | cpio -idmv
+    rpm2cpio ../kernel-modules-$kernel_version.rpm | cpio -idmv
+
+    sudo cp -ar ./usr/* "$ROOTFS_DIR"/usr
+    sudo cp -ar ./lib/* "$ROOTFS_DIR"/lib
+
+    # Generate modules.dep and map files, so modules autoload on first boot
+    depmod -b "$ROOTFS_DIR" $kernel_version
+
+    # Create a directory tree similar to the kernel source tree so we can reuse some functions
+    # like cmd_build_vboot and cmd_deploy_vboot
+    mkdir -p arch/arm64/boot/dts
+    cp $ROOTFS_DIR/lib/modules/$kernel_version/vmlinuz arch/arm64/boot/Image.gz
+    gunzip arch/arm64/boot/Image.gz
+    cp -fr $ROOTFS_DIR/lib/modules/$kernel_version/dtb/* arch/arm64/boot/dts/
+
+    create_fit_image
+
+    cd - > /dev/null
+
+    export CB_KERNEL_PATH=./tmpdir
+    cmd_build_vboot
+    cmd_deploy_vboot
+
+    sudo rm -rf ./tmpdir
+
+}
+
+cmd_deploy_fedora()
+{
+    if [ ! -f "$IMAGE" ] && [ "$IMAGE" != "" ] || [ -z "$IMAGE" ]; then
+        echo "Error: $IMAGE not found! Downloading the default fedora image."
```

Comment:
I'd remove "Error $IMAGE not found!" and just put "Downloading the default fedora image" as if you don't pass any image is confusing to see an error. Not passing an image is just fine.